libraries/c_sdk/standard/mqtt/test/unit/iot_tests_mqtt_subscription.c

```diff
@@ -340,26 +426,27 @@ TEST_GROUP_RUNNER( MQTT_Unit_Subscription )
  */
 TEST( MQTT_Unit_Subscription, ListInsertRemove )
 {
-    _mqttSubscription_t node1;
-    _mqttSubscription_t node2;
-    _mqttSubscription_t node3;
-
-    ( void ) memset( &node1, 0x00, sizeof( _mqttSubscription_t ) );
-    ( void ) memset( &node2, 0x00, sizeof( _mqttSubscription_t ) );
-    ( void ) memset( &node3, 0x00, sizeof( _mqttSubscription_t ) );
-
-    IotListDouble_InsertHead( &( _pMqttConnection->subscriptionList ),
-                              &( node1.link ) );
-    IotListDouble_InsertHead( &( _pMqttConnection->subscriptionList ),
-                              &( node2.link ) );
-    IotListDouble_InsertHead( &( _pMqttConnection->subscriptionList ),
-                              &( node3.link ) );
-
-    IotListDouble_Remove( &( node1.link ) );
-    IotListDouble_Remove( &( node2.link ) );
-    IotListDouble_Remove( &( node3.link ) );
-
-    TEST_ASSERT_EQUAL_INT( true, IotListDouble_IsEmpty( &( _pMqttConnection->subscriptionList ) ) );
+    int8_t index1 = -1;
+    int8_t index2 = -1;
+    int8_t index3 = -1;
+
+    index1 = IotMqtt_GetFreeIndexInSubscriptionArray( connToContext[ contextIndex ].subscriptionArray );
+
+    TEST_ASSERT_NOT_EQUAL( -1, index1 );
+
+    index2 = IotMqtt_GetFreeIndexInSubscriptionArray( connToContext[ contextIndex ].subscriptionArray );
+
+    TEST_ASSERT_NOT_EQUAL( -1, index2 );
+
+    index3 = IotMqtt_GetFreeIndexInSubscriptionArray( connToContext[ contextIndex ].subscriptionArray );
+
+    TEST_ASSERT_NOT_EQUAL( -1, index3 );
+
+    IotMqtt_RemoveSubscription( connToContext[ contextIndex ].subscriptionArray, index1 );
+    IotMqtt_RemoveSubscription( connToContext[ contextIndex ].subscriptionArray, index2 );
+    IotMqtt_RemoveSubscription( connToContext[ contextIndex ].subscriptionArray, index3 );
```

Comment:
Is this step necessary as we are not assigning anything to the subscription?

Reply:
I have created this test to just check whether the insertion and removal in subscription array is happening successfully.